tests/Mock/MockObject1.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Aeviiq\Tests\Collection\Mock;
+
+final class MockObject1 implements MockObjectInterface
```

Comment:
You could use anonymous classes for these and made a private method which will return this instance. This way these excess files don't need to be created.  For their allowedInstance() you could use the stdClass::class

Reply:
Removed the mock objects and collection. I left the interface there so I can use the functions present on the interface inside of the tests (i.e. in the map function I use the getText method and in uasort I use the getInt method)